internal/ssm/source.go

```diff
@@ -45,6 +45,19 @@ func (s ssmInstallerSource) GetSSMInstaller(ctx context.Context) (io.ReadCloser,
 	return obj, nil
 }
 
+func (s ssmInstallerSource) GetSSMInstallerSignature(ctx context.Context) (io.ReadCloser, error) {
+	endpoint, err := buildSSMURL(s.region)
+	if err != nil {
+		return nil, err
+	}
+	obj, err := util.GetHttpFileReader(ctx, endpoint+".sig")
+	if err != nil {
+		obj.Close()
```

Comment:
does the obj need to be closed when there is an error? maybe we need to check the implementation, but generally when a "constructor" builds something that needs to be closed, that thing should only be closed when the constructor succeeds. Else, you might be. it might blow up since you could be getting a nil object